### PR TITLE
CNI gets networkconfig from CNS 

### PR DESCRIPTION
**Pull request by tamilmani1989**
**What this PR does / why we need it**:
This PR adds support for CNI to query and get network config based on podname and namespace. Also added an option in cns to not to start cnm by default.
 
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: client/cnsclient/cnsclient.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+const (
+	defaultCnsURL = "http://localhost:10090"
+)
+
+// NewCnsClient create a new cns client.
+func NewCnsClient(url string) (*CNSClient, error) {
+	if url == "" {
+		url = defaultCnsURL
+	}
+	return &CNSClient{
```
Comment (saiyan86) [inline client/cnsclient/cnsclient.go]: space

File: cni/network/network.go [hunk 2]
```diff
@@ -6,20 +6,25 @@ package network
...
 	cniSkel "github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
 	cniTypesCurr "github.com/containernetworking/cni/pkg/types/current"
 )
 
 const (
 	// Plugin name.
-	name = "azure-vnet"
+	name         = "azure-vnet"
+	namespacekey = "K8S_POD_NAMESPACE"
```
Comment (saiyan86) [inline cni/network/network.go]: rename to namespaceKey?

File: cni/network/network.go [hunk 3]
```diff
@@ -6,20 +6,25 @@ package network
...
+	podnamekey   = "K8S_POD_NAME"
```
Comment (saiyan86) [inline cni/network/network.go]: podNameKey?

File: cni/plugin.go [hunk 4]
```diff
@@ -168,6 +169,20 @@ func (plugin *Plugin) GetEndpointID(args *cniSkel.CmdArgs) string {
...
+	argsMap := make(map[string]interface{})
+
+	data := strings.Split(args, ";")
+	for _, pair := range data {
+		items := strings.Split(pair, "=")
+		if len(items) > 1 {
+			argsMap[items[0]] = items[1]
+		}
+	}
+
```
Comment (saiyan86) [inline cni/plugin.go]: Is args always guaranteed to have this format? "key0=val0; key1=val1"

File: cns/service/main.go [hunk 5]
```diff
@@ -211,28 +194,59 @@ func main() {
...
+			if err != nil {
+				fmt.Printf("Failed to start network plugin, err:%v.\n", err)
+				return
+			}
+		}
+
+		ipamPlugin.SetOption(acn.OptEnvironment, environment)
+		ipamPlugin.SetOption(acn.OptAPIServerURL, url)
+		ipamPlugin.SetOption(acn.OptIpamQueryInterval, ipamQueryInterval)
+		if ipamPlugin != nil {
```
Comment (saiyan86) [inline cns/service/main.go]: Isn't ipamPlugin guaranteed to be valid at this point? We already catched err at line 217.

File: client/cnsclient/cnsclient.go (hunk 1 above)
Comment (tamilmani1989) [reply client/cnsclient/cnsclient.go]: ok

File: cni/network/network.go (hunk 2 above)
Comment (tamilmani1989) [reply cni/network/network.go]: ok

File: cni/network/network.go (hunk 3 above)
Comment (tamilmani1989) [reply cni/network/network.go]: ok

File: cni/plugin.go (hunk 4 above)
Comment (tamilmani1989) [reply cni/plugin.go]: yes

File: cns/service/main.go (hunk 5 above)
Comment (tamilmani1989) [reply cns/service/main.go]: not needed for netplugin either..will change both

**Pull request merged**